File: frontend/src/api/roadSections.ts
/**
 * API functions for road sections (участки дороги) section
 */

import { request, API_VERSION } from './client'

const API_BASE = `/api/${API_VERSION}/road-sections`

// --- Types ---

export interface RoadSectionRow {
  road_section_id: number
  road_section_name: string | null
  length_km: number | null
  passport_volume: number | null
  sidewalk_passport_volume: number | null
}

export interface RoadSectionsListResponse {
  rows: RoadSectionRow[]
  total: number
}

// --- API Functions ---

/**
 * Get list of road sections with optional search
 */
export async function getRoadSections(search?: string): Promise<RoadSectionsListResponse> {
  const queryParams = new URLSearchParams()
  if (search) queryParams.set('search', search)
  
  const queryString = queryParams.toString()
  const url = queryString ? `${API_BASE}?${queryString}` : API_BASE
  
  return await request<RoadSectionsListResponse>(url)
}

File: app/backend/schemas/road_sections.py
"""Schemas for road sections (участки дороги) section."""

from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class _BaseSchema(BaseModel):
    """Base schema with optimized serialization settings."""
    model_config = ConfigDict(
        extra="ignore",
        from_attributes=True,
        ser_json_inf_nan="constants",
    )


class RoadSectionRow(_BaseSchema):
    """Single road section row."""
    road_section_id: int
    road_section_name: Optional[str]
    length_km: Optional[float]
    passport_volume: Optional[float]
    sidewalk_passport_volume: Optional[float]


class RoadSectionsListResponse(_BaseSchema):
    """Response with list of road sections."""
    rows: List[RoadSectionRow]
    total: int
